physical/mysql/mysql.go

```diff
@@ -251,14 +256,18 @@ func NewMySQLClient(conf map[string]string, logger log.Logger) (*sql.DB, error)
 	}
 
 	dsnParams := url.Values{}
-	tlsCaFile, ok := conf["tls_ca_file"]
-	if ok {
+	tlsCaFile, tls_ok := conf["tls_ca_file"]
+	if tls_ok {
 		if err := setupMySQLTLSConfig(tlsCaFile); err != nil {
 			return nil, errwrap.Wrapf("failed register TLS config: {{err}}", err)
 		}
 
 		dsnParams.Add("tls", mysqlTLSKey)
 	}
+	_, pt_ok := conf["plaintext_connection_allowed"]
+	if !(pt_ok || tls_ok) {
+		return nil, ErrPlaintextTransmission
```

Comment:
Should this instead be a logged warning for 1.4/1.5, and a requirement/error for 1.6?

Reply:
We could potentially do that, yes. Up to others as I'm not super used to this kind of dev cycle yet :)